db/system_keyspace.cc

```diff
@@ -372,6 +372,7 @@ schema_ptr system_keyspace::built_indexes() {
                 {"schema_version", uuid_type},
                 {"tokens", set_type_impl::get_instance(utf8_type, true)},
                 {"supported_features", utf8_type},
+                {"raft_server_id", uuid_type},
```

Comment:
System tables *can* be altered if done with care. The procedure is more involved if the table you are modifying is a distributed table. For local tables it is much simpler. The main thing is that have to make sure you are not using the new field until all nodes in the cluster are on the new version. We usually use a feature flag for this.
@bhalevy recently did a change to a local system table, you can check what he did here https://github.com/scylladb/scylladb/pull/11674/commits/2f49eebb048bd5be465706496579e4aeb0083890.

With system tables originating from C* there is also the concern of compatibility with C*. Adding a new field is fine from my POV, but I'd like to hear @avikivity's opinion on that too.

Reply:
@kbr- so the right thing to do is to only add the new field in do_update_system_peers_table() if all nodes in the cluster support raft - i.e. the feature bit was set.

Reply:
It's not safe without extra care, see https://github.com/scylladb/scylladb/issues/11660

Reply:
Botond's commit takes care of this.

Reply:
I think it's okay to add columns to an existing Cassandra derived table.

Reply:
the group id is unique, because the server can have many groups. if you mean raft server id is different from host id, it's because host id is preserved during replacenode op, and raft ids can't be reused.

Reply:
Changed to not store raft server id unless the feature is enabled in the entire cluster.

Reply:
Add schema_version_offset to system.peers